File: app/src/main/java/com/feriantes4dawin/feriavirtualmovil/data/LoginDataSource.java
package com.feriantes4dawin.feriavirtualmovil.data;

import android.util.Log;

import com.feriantes4dawin.feriavirtualmovil.data.model.LoggedInUser;
import java.io.IOException;

/**
 * Class that handles authentication w/ login credentials and retrieves user information.
 */
public class LoginDataSource {

    public Result<LoggedInUser> login(String username,String password) {

        Result<LoggedInUser> r = new Result<LoggedInUser>();

        try {
            // TODO: handle loggedInUser authentication
            LoggedInUser fakeUser = new LoggedInUser(java.util.UUID.randomUUID().toString(), "Usuario promedio");
            return r.new Success<LoggedInUser>(fakeUser);

        } catch (Throwable t) {

            return r.new Error<LoggedInUser>(
                new IOException("Error logging in", t)
            );

        }

    }

    public void logout() {
        // TODO: revoke authentication
    }
}

File: app/src/main/java/com/feriantes4dawin/feriavirtualmovil/data/LoginDataSource.kt
package com.feriantes4dawin.feriavirtualmovil.data

import com.feriantes4dawin.feriavirtual.data.model.LoggedInUser
import java.io.IOException

/**
 * Class that handles authentication w/ login credentials and retrieves user information.
 */
class LoginDataSource {

    fun login(username: String, password: String): _root_ide_package_.com.feriantes4dawin.feriavirtualmovil.data.Result<LoggedInUser> {

        try {
            // TODO: handle loggedInUser authentication
            val fakeUser = LoggedInUser(java.util.UUID.randomUUID().toString(), "Usuario promedio")
            return _root_ide_package_.com.feriantes4dawin.feriavirtualmovil.data.Result.Success(
                fakeUser
            )
        } catch (e: Throwable) {
            return _root_ide_package_.com.feriantes4dawin.feriavirtualmovil.data.Result.Error(
                IOException("Error logging in", e)
            )
        }

    }

    fun logout() {
        // TODO: revoke authentication
    }
}